core/src/com/unciv/ui/utils/ZoomableScrollPane.kt

```diff
@@ -11,10 +11,6 @@ open class ZoomableScrollPane: ScrollPane(null) {
     var continousScrollingX = false
 
     init{
-        // Remove the existing inputListener
```

Comment:
There's no reason to put this in addZoomListeners... That makes the init only call to an external function.

Reply:
The compile will give a warning "Calling non-final function removeListener in constructor"
Moving the method "removeListener()" to this will get rid of this warning because addZoomListeners() is private final fun

Reply:
There's no problem with calling a non final fun in the constructor. I get why it's a warning - so you won't override one of the base functions and interfere with the internals of the class, but it's really not relevant here.

In general, I see you doing a lot of like 'return if () x else y' replacing 'if () return x else return y' and I assume these are Android studio suggestions. Not all their suggestions are really necessary... And specifically, return if else is to my eyes less immediately readable than if return else return...

Reply:
OK, about this, your mean is that I should go back to the older branch, is it?
By the way, all method named "equals()" also should be edit because it may conflict with equals() in **"class Any"**, equals() in **"class Any"** is always used in operator "=="

Reply:
No, I'm saying that it doesn't really matter that much, so either way is fine.